app/models/node.rb

```diff
@@ -110,4 +111,13 @@ def destroy_attachments
     attachments_dir = Attachment.pwd.join(self.id.to_s)
     FileUtils.rm_rf attachments_dir if File.exists?(attachments_dir)
   end
+
+  def parent_node
+    if self.parent.nil? ||
+      !(self.parent.type_id == Node::Types::DEFAULT ||
+        self.parent.type_id == Node::Types::HOST)
+
+      errors.add(:base, 'Invalid parent.')
```

Comment:
Since we're checking both "invalid parent" (nil) and parent assigned to a user node, should we present a different error message? I'm not sure if it's worth it.